src/LandIce/evaluators/LandIce_BasalFrictionCoefficient_Def.hpp

```diff
@@ -242,115 +303,130 @@ KOKKOS_INLINE_FUNCTION
           }
+          Nval = g*KU::max(rho_i*thickness+KU::min(rho_w*bed_topo,0.0),0.0);
+        }
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::HYDROSTATIC_AT_NODES:
+        if(nodal)
+          Nval = g* KU::max(rho_i*thickness_field(cell,ipt)+KU::min(rho_w*bed_topo_field(cell,ipt),0.0),0.0);
+        else {
+          Nval = 0;
+          for (unsigned int node=0; node<numNodes; ++node)
+            Nval += g*KU::max(rho_i*thickness_field(cell,node)+KU::min(rho_w*bed_topo_field(cell,node),0.0),0.0)*BF(cell,node,ipt);
         }
+        break;
       }
-    break;
-
-    case EXP_GIVEN_FIELD:
-      if(nodal || interpolate_then_exponentiate) {
-        if (is_given_field_param) {
-          for (unsigned int ipt=0; ipt<dim; ++ipt)
-            beta(cell,ipt) = std::exp(given_field_param(cell,ipt));
-        } else {
-          for (unsigned int ipt=0; ipt<dim; ++ipt)
-            beta(cell,ipt) = std::exp(given_field(cell,ipt));
+
+      switch (mu_type) {
+      case FIELD_TYPE::FIELD:
+        muValue = muField(cell,ipt);
+        break;
+      case FIELD_TYPE::EXPONENT_OF_FIELD_AT_NODES:
+        if(nodal)
+          muValue = std::exp(muField(cell,ipt));
+        else {
+          muValue = 0;
+          for (unsigned int node=0; node<numNodes; ++node)
+            muValue += std::exp(muField(cell,node))*BF(cell,node,ipt);
         }
-      } else {
-        if (is_given_field_param) {
-          unsigned int exp_dim = is_side_equation ? numQPs : dim;
-          for (unsigned int ipt=0; ipt<exp_dim; ++ipt) {
-            beta(cell,ipt) = 0;
-            for (unsigned int node=0; node<numNodes; ++node)
-              beta(cell,ipt) += std::exp(given_field_param(cell,node))*BF(cell,node,ipt);  
-          }
-        } else {
-          unsigned int exp_dim = is_side_equation ? numQPs : dim;
-          for (unsigned int ipt=0; ipt<exp_dim; ++ipt) {
-            beta(cell,ipt) = 0;
+        break;
+      case FIELD_TYPE::EXPONENT_OF_FIELD:
+        muValue = std::exp(muField(cell,ipt));
+        break;
+      case FIELD_TYPE::UNIFORM:
+        muValue = mu;
+        break;
+      }
+
+      if (power == 1.0)
+        beta(cell,ipt) = muValue * Nval;
+      else
+        beta(cell,ipt) = muValue * Nval * std::pow (u_norm(cell,ipt), power-1.0);
+
+      if (beta_type == REGULARIZED_COULOMB) {
+        ParamScalarT lambdaValue;
+        switch (lambda_type) {
+        case FIELD_TYPE::FIELD:
+          lambdaValue = lambdaField(cell,ipt);
+          break;
+        case FIELD_TYPE::EXPONENT_OF_FIELD_AT_NODES:
+          if(nodal)
+            lambdaValue = std::exp(lambdaField(cell,ipt));
+          else {
+            lambdaValue = 0;
             for (unsigned int node=0; node<numNodes; ++node)
-              beta(cell,ipt) += std::exp(given_field(cell,node))*BF(cell,node,ipt);
+              lambdaValue += std::exp(lambdaField(cell,node))*BF(cell,node,ipt);
           }
+          break;
+        case FIELD_TYPE::EXPONENT_OF_FIELD:
```

Comment:
Same consideration as for mu: we could keep just one between EXPONENT_OF_FIELD and EXPONENT_OF_FIELD_AT_NODES, and always do interp(exp(f_nodal)).

Reply:
In this case we could get rid of it, as we are not breaking any backward compatibility (lambda has never been used as exponent of a field)